### PR TITLE
TGP-2698: CYA New UKIMS Number

### PR DESCRIPTION
**Pull request by yohan-oc**
# Checklist for developers before reviewing
 - [x] I have merged in main
 - [x] I have assigned the PR to me
 - [x] I ran the code with a clean compile test and ensured there are no warnings or deprecations
 - [x] I have run `sbt prePR` to run all tests, format and check the code coverage
 - [x] I have written unit tests to cover any code that I wrote, where necessary
 - [x] I have linked tickets to any TODOs I have written
 - [x] I have run the microservices with the latest SM2 config
 - [x] I have checked all the Acceptance Criteria pass
 - [x] I have run the pipeline journey tests
 - [x] I have run the local journey tests
 - [x] I have spoken to the QAs about any changes that may need to be made to the journey tests OR no changes to the journey tests are necessary
 - [x] I have moved the ticket to Review in Jira
 - [x] I have left a message in tgp-prs on slack to alert other Devs that this ticket is ready for review
 - [x] I have checked my PR is not in draft mode

# Checklist for reviewer 1
- [x] I have added an emoji to the relevant message in tgp-prs to show I am reviewing
- [x] I have checked out the branch
- [x] I have run the microservices with the latest SM2 config
- [x] I have checked all the Acceptance Criteria pass
- [x] I reviewed the code and left comments where necessary
- [x] I have left a message on the relevant tgp-prs message to show that I have finished reviewing

# Checklist for reviewer 2
- [ ] I have added an emoji to the relevant message in tgp-prs to show I am reviewing
- [ ] I have checked out the branch
- [ ] I have run the microservices with the latest SM2 config
- [ ] I have checked all the Acceptance Criteria pass
- [ ] I reviewed the code and left comments where necessary
- [ ] I have left a message on the relevant tgp-prs message to show that I have finished reviewing

# Checklist for developers before merging
 - [x] The PR has been approved by 2 other developers
 - [x] I have merged in main
 - [x] I have run `sbt preMerge` to run all tests, check formatting and code coverage
 - [x] I have run the microservices with the latest SM2 config
 - [x] I have checked all the Acceptance Criteria pass
 - [x] I have checked that the QAs have merged in any changes to the journey tests
 - [x] I have run the pipeline journey tests
 - [x] I have run the local journey tests

# Reminders for developers after merging
 1. Wait for pipeline to pass
 2. Move ticket to QA column in Jira
 3. Leave message in tgp-qa on slack to alert QAs that this ticket is ready for testing

### CODE REVIEW

File: app/controllers/CyaNewUkimsNumberController.scala [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      }
+  }
+
+  def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    (for {
+      ukimsNumber                <- handleValidateError(TraderProfile.validateUkimsNumber(request.userAnswers))
+      oldTraderProfile           <- traderProfileConnector.getTraderProfile(request.eori)
+      newTraderProfile           <- Future.successful(oldTraderProfile.copy(ukimsNumber = ukimsNumber))
+      _                           = auditService.auditMaintainProfile(oldTraderProfile, newTraderProfile, request.affinityGroup)
+      _                          <- traderProfileConnector.submitTraderProfile(newTraderProfile, request.eori)
```
Comment (besscerule) [inline app/controllers/CyaNewUkimsNumberController.scala:75]: Do we always have to call submit? Is there a user journey where the user can get to this page and submit the previous ukims number? In the other CYAs we only submit if necessary.

File: test/navigation/NavigatorSpec.scala [hunk 2]
```diff
@@ -3010,6 +3013,18 @@ class NavigatorSpec extends SpecBase with BeforeAndAfterEach {
           ) mustBe routes.DownloadRequestSuccessController.onPageLoad()
         }
       }
+
+      "in New UKIMS Number update Journey" - {
+        "must go from UkimsNumberUpdatePage to CyaMaintainProfilePage" in {
```
Comment (besscerule) [inline test/navigation/NavigatorSpec.scala]: typo in test

Comment (yohan-oc) [reply test/navigation/NavigatorSpec.scala:3018]: Here, we need to call `submitTraderProfile` to set `eoriChange` to `false`.

File: app/controllers/CyaNewUkimsNumberController.scala (hunk 1 above)
Comment (yohan-oc) [reply app/controllers/CyaNewUkimsNumberController.scala:75]: Here, we need to call submitTraderProfile to set eoriChange to false.

File: test/navigation/NavigatorSpec.scala (hunk 2 above)
Comment (yohan-oc) [reply test/navigation/NavigatorSpec.scala]: thank you  :)

**Review by besscerule — APPROVED** [review]

**Review by pauloncast — APPROVED** [review]

**Pull request merged**